plugins/00_dokku-standard/subcommands/run

```diff
@@ -6,12 +6,38 @@ source "$PLUGIN_AVAILABLE_PATH/config/functions"
 
 dokku_run_cmd() {
   declare desc="runs command in container based on app image"
-  declare cmd="run" APP="$2"
+  declare cmd="run" APP=""
+  shift
+
+  declare -a RUN_ENV
+  RUN_ENV=()
+  while [[ $# -gt 0 ]]; do
+    case $1 in
+      -e=* | --env=*)
+        local arg=$(printf "%s" "$1" | sed -E 's/(^-e=)|(^--env=)//g')
+        RUN_ENV+=("$arg")
+        shift
+        ;;
+      -e | --env)
+        if [[ ! $2 ]]; then
+          dokku_log_warn "expected $1 to have an argument"
+          break
+        fi
+        RUN_ENV+=("$2")
+        shift 2
+        ;;
+      *)
+        APP="$1"
+        shift
+        break
+        ;;
+    esac
+  done
+
   verify_app_name "$APP"
-  shift 2
 
   local DOKKU_SCHEDULER=$(get_app_scheduler "$APP")
-  plugn trigger scheduler-run "$DOKKU_SCHEDULER" "$APP" "$@"
+  plugn trigger scheduler-run "$DOKKU_SCHEDULER" "$APP" "${#RUN_ENV[@]}" "${RUN_ENV[@]}" "$@"
```

Comment:
Is there a better way to get this information to the scheduler? This is the best I could come up with that meets my goals of minimizing changes and processing inside the scheduler, but I have a feeling there's a better way.

Reply:
Given that there is only one scheduler in existence atm - docker-local - it's fine for us to change the interface in the next minor.

Maybe we pass a json blob of data for the run-env?

I do wonder what other types of info we'd pass to the `scheduler-run` trigger, and whether there is a better way generally to structure it. Assuming the docker-local plugin was in a higher-level language, do you have any suggestions on how this might change?

Reply:
Hmm, yeah I feel like json would could work. Any language triggering plugins or the scheduler itself would be written in would have robust support for json at the very least. I can foresee in the maybe future separating the app environment from the scheduler environment (or parts thereof), so it might be nice to use a format that has room for that kind of growth. I'm excited for the time when more and more things get added to it and someone finally blows through ARG_MAX :-p

On the other hand, this is really one call site and one implementation, so worrying about it now might be premature. Maybe leave it as it and revisit it when another scheduler is added or the next time new arguments need to be added to the `scheduler-run` trigger?

Reply:
If it works as is, thats fine by me.

Aside: http://dokku.viewdocs.io/dokku/advanced-usage/deployment-tasks/

Reply:
I'm not quite following the relationship between deployment tasks and this. I do see that there are a number of JSON-in-bash utilities in dokku already that are exercised in that plugin, so threading this info through the trigger as JSON would be easier than I thought. Is there something else to deployment-tasks I should be thinking about?

Reply:
Ah, I meant to link to the [kubernetes](https://github.com/dokku/dokku-scheduler-kubernetes) plugin - we also have a [nomad](https://github.com/dokku/dokku-scheduler-nomad) one - as scheduler implementations.

Reply:
In the context of having alternative schedulers already implemented, is that something that would change the implementation? If not, that's fine.